sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowTablesExtendedExec.scala

```diff
@@ -0,0 +1,194 @@
+    val tableType = if (table.properties().containsKey(TableCatalog.PROP_EXTERNAL)) {
+      CatalogTableType.EXTERNAL
+    } else {
+      CatalogTableType.MANAGED
+    }
+    results.put("Type", tableType.name)
+
+    CatalogV2Util.TABLE_RESERVED_PROPERTIES
+      .filterNot(_ == TableCatalog.PROP_EXTERNAL)
+      .foreach(propKey => {
+        if (table.properties.containsKey(propKey)) {
+          results.put(propKey.capitalize, table.properties.get(propKey))
+        }
+      })
+
+    val properties =
+      conf.redactOptions(table.properties.asScala.toMap).toList
+        .filter(kv => !CatalogV2Util.TABLE_RESERVED_PROPERTIES.contains(kv._1))
+        .sortBy(_._1).map {
+        case (key, value) => key + "=" + value
+      }.mkString("[", ",", "]")
+    if (table.properties().isEmpty) {
+      results.put("Table Properties", properties.mkString("[", ", ", "]"))
+    }
+
+    // Partition Provider & Partition Columns
+    var partitionColumns = new StructType()
+    if (table.supportsPartitions && table.asPartitionable.partitionSchema().nonEmpty) {
+      partitionColumns = table.asPartitionable.partitionSchema()
+      results.put("Partition Provider", "Catalog")
+      results.put("Partition Columns", table.asPartitionable.partitionSchema().map(
+        field => quoteIdentifier(field.name)).mkString("[", ", ", "]"))
+    }
+
+    if (table.schema().nonEmpty) {
+      val dataColumns = table.schema().filterNot(partitionColumns.contains)
+      results.put("Schema", StructType(dataColumns ++ partitionColumns).treeString)
+    }
+
+    results.map { case (key, value) =>
+      if (value.isEmpty) key else s"$key: $value"
+    }.mkString("", "\n", "")
+  }
+}
+
+/**
+ * Physical plan node for showing tables with partition, Show the information of partitions.
+ */
+case class ShowTablePartitionExec(
+    output: Seq[Attribute],
+    catalog: TableCatalog,
+    tableIndent: Identifier,
+    table: SupportsPartitionManagement,
+    partSpec: ResolvedPartitionSpec) extends V2CommandExec with LeafExecNode {
+  override protected def run(): Seq[InternalRow] = {
+    val rows = new ArrayBuffer[InternalRow]()
+    val information = getTablePartitionDetails(tableIndent,
+      table, partSpec)
+    rows += toCatalystRow(tableIndent.namespace.quoted,
+      tableIndent.name(), false, s"$information\n")
+
+    rows.toSeq
+  }
+
+  private def getTablePartitionDetails(
+      tableIdent: Identifier,
+      partitionTable: SupportsPartitionManagement,
+      partSpec: ResolvedPartitionSpec): String = {
+    val results = new mutable.LinkedHashMap[String, String]()
+
+    // "Partition Values"
+    val partitionSchema = partitionTable.partitionSchema()
+    val (names, ident) = (partSpec.names, partSpec.ident)
+    val partitionIdentifiers = partitionTable.listPartitionIdentifiers(names.toArray, ident)
+    if (partitionIdentifiers.isEmpty) {
+      throw QueryCompilationErrors.notExistPartitionError(tableIdent, ident, partitionSchema)
```

Comment:
This can only happen when the partition is deleted right after we resolve to `ResolvedPartitionSpec`?

Reply:
When a partition does not exist, it triggers this exception
Because we didn't actually retrieve this partition from the meta during `UnresolvedPartitionSpec` to `ResolvedPartitionSpec`.
https://github.com/apache/spark/blob/1359c1327345efdf9a35c46a355b5f928ac33e6d/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolvePartitionSpec.scala#L42-L74

According to my understanding, it only detects whether the `partition key` is valid and does some normalized of the `partition key`, the existence of the `partition value` is not verified.
If my understanding is incorrect, please point it out.

Reply:
ah I see!